pkg/api/pkgecosystem/ecosystem.go

```diff
@@ -0,0 +1,64 @@
+// The pkgecosystem package defines the open source ecosystems supported by Package Analysis.
+package pkgecosystem
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Ecosystem represents an open source package ecosystem from which packages can be downloaded.
+//
+// It implements encoding.TextUnmarshaler and encoding.TextMarshaler so it can
+// be used with flag.TextVar.
+type Ecosystem string
+
+const (
+	None      Ecosystem = ""
+	CratesIO  Ecosystem = "crates.io"
+	NPM       Ecosystem = "npm"
+	Packagist Ecosystem = "packagist"
+	PyPI      Ecosystem = "pypi"
+	RubyGems  Ecosystem = "rubygems"
+)
+
+// FlagUsage is a pregenerated string for use as usage text in flag.TextVar for
+// an ecosystem flag.
+const FlagUsage = "package ecosystem. Can be npm, pypi, rubygems, packagist, or crates.io"
```

Comment:
This probably shouldn't be in the public API

Reply:
How so?

I think it makes sense, for example Package Detection has a CLI tool that accepts "ecosystem" as an argument. Seems reasonable to have a ready to use usage string.

Reply:
It seems like it would be more typical for an API to export the list of supported ecosystems and have something very UI specific like a usage string be internal.

Reply:
PTAL, I've generalized it more (although there is now more code).